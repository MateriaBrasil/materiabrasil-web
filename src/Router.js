import React from 'react'
import Route from 'react-router-dom/Route'
import Switch from 'react-router-dom/Switch'

import Materials from './materials/Materials'
import Material from './material/Material'

export default () => (
  <Switch>
    <Route exact path="/" component={Materials} />
    <Route exact path="/:materialId" component={Material} />
  </Switch>
)
