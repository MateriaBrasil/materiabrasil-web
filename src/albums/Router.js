import React from 'react'
import Route from 'react-router-dom/Route'

import renderInfo from './renderInfo'
import renderPremium from '../payments/renderPremium'
import renderPay from '../payments/renderPay'
import renderSuccess from '../payments/renderSuccess'
import renderMember from './renderMember'

export default props => (
  <div>
    <Route path="/albums/:id" render={renderInfo(props)} />
    <Route exact path="/albums/:id/premium" render={renderPremium(props)} />
    <Route exact path="/albums/:id/pay" render={renderPay(props)} />
    <Route exact path="/albums/:id/success" render={renderSuccess(props)} />
    <Route exact path="/albums/:id/member" render={renderMember(props)} />
  </div>
)
