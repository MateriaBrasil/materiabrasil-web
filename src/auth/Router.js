import React from 'react'
import Route from 'react-router-dom/Route'
import Switch from 'react-router-dom/Switch'

import renderSignIn from './signIn/render'
import renderSignUp from './signUp/render'
import renderResetPassword from './resetPassword/render'

export default props => (
  <Switch>
    <Route exact path="/auth/sign-in" render={renderSignIn(props)} />
    <Route exact path="/auth/sign-up" render={renderSignUp(props)} />
    <Route
      exact
      path="/auth/reset-password"
      render={renderResetPassword(props)}
    />
  </Switch>
)
