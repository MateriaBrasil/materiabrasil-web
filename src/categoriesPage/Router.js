import React from 'react';
import Route from 'react-router-dom/Route';

import renderCategorySingle from './info';

export default props => (
  <div>
    <Route path="/categories/:id" render={renderCategorySingle(props)} />
  </div>
);
