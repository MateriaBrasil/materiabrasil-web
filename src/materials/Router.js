import React from 'react';
import Route from 'react-router-dom/Route';

import renderReviews from '../reviews/render';
import renderList from './renderList';
import renderInfo from './renderInfo';
import renderEdit from './renderEdit';
import renderCategories from '../categories/render';
import renderImageUpload from './renderImageUpload';
import renderQuestionnaires from '../questionnaires/renderList';
import renderAlbums from '../albums/renderList';
import renderNew from '../albums/renderNew';

export default props => (
  <div>
    <Route exact path="/materials" render={renderList(props)} />
    <Route exact path="/materials/:id/edit" render={renderEdit(props)} />
    <Route
      exact
      path="/materials/:id/categories"
      render={renderCategories(props)}
    />
    <Route path="/materials/:id" render={renderInfo(props)} />
    <Route path="/materials/:id/reviews" render={renderReviews(props)} />
    <Route exact path="/materials/:id/albums" render={renderAlbums(props)} />
    <Route exact path="/materials/:id/albums/new" render={renderNew(props)} />
    <Route
      exact
      path="/materials/:id/coverImage"
      render={renderImageUpload({
        ...props,
        width: 1920,
        height: 640,
        attributeName: 'coverImageUrl',
      })}
    />
    <Route
      exact
      path="/materials/:id/listImage"
      render={renderImageUpload({
        ...props,
        width: 720,
        height: 540,
        attributeName: 'listImageUrl',
      })}
    />
    <Route
      exact
      path="/materials/:id/highlightImage"
      render={renderImageUpload({
        ...props,
        width: 1920,
        height: 720,
        attributeName: 'highlightImageUrl',
      })}
    />
    <Route
      exact
      path="/materials/:id/questionnaires"
      render={renderQuestionnaires({ ...props, aboutType: 'Material' })}
    />
  </div>
);
