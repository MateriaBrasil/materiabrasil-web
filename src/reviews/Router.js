import React from 'react'
import Route from 'react-router-dom/Route'
import Switch from 'react-router-dom/Switch'

import renderNew from './renderNew'

export default props => (
  <Switch>
    <Route exact path="/:id/reviews/new" render={renderNew(props)} />
  </Switch>
)
