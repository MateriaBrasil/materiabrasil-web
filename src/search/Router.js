import React from 'react'
import Route from 'react-router-dom/Route'
import Switch from 'react-router-dom/Switch'

import renderRoot from './renderRoot'
import renderList from './renderList'

export default props => (
  <Switch>
    <Route exact path="/search" render={renderRoot(props)} />
    <Route exact path="/search/:term" render={renderList(props)} />
  </Switch>
)
