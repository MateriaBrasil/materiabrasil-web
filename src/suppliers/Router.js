import React, { Fragment } from 'react'
import Route from 'react-router-dom/Route'

import renderNewMaterial from '../materials/renderNew'
import renderNewMessage from '../messages/renderNew'
import renderSuccessMessage from '../messages/renderSuccess'

import renderNew from './renderNew'
import renderEdit from './renderEdit'
import renderInfo from './renderInfo'
import renderImageUpload from './renderImageUpload'

export default props => {
  return (
    <Fragment>
      <Route path="/suppliers/:id(\d+)" render={renderInfo(props)} />
      <Route exact path="/suppliers/new" render={renderNew(props)} />
      <Route
        exact
        path="/suppliers/:id/materials/new"
        render={renderNewMaterial(props)}
      />
      <Route
        exact
        path="/suppliers/:id/messages/new"
        render={renderNewMessage({ ...props, toType: 'Supplier' })}
      />
      <Route
        exact
        path="/suppliers/:id/messages/success"
        render={renderSuccessMessage({ ...props })}
      />
      <Route
        exact
        path="/suppliers/:id/avatar"
        render={renderImageUpload(props)}
      />
      <Route exact path="/suppliers/:id/edit" render={renderEdit(props)} />
    </Fragment>
  )
}
