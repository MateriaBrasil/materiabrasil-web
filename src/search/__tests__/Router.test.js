import React from 'react'
import renderer from 'react-test-renderer'
import { MemoryRouter } from 'react-router-dom'

import Router from '../Router'

jest.mock('../renderList', () => props => routeProps => <div>renderList</div>)
jest.mock('../renderRoot', () => props => routeProps => <div>renderRoot</div>)

describe('with /search route', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <MemoryRouter initialEntries={['/search']}>
          <Router />
        </MemoryRouter>,
      )
      .toJSON()

    expect(tree).toMatchSnapshot()
  })
})

describe('with /search/:term route', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <MemoryRouter initialEntries={['/search/:term']}>
          <Router />
        </MemoryRouter>,
      )
      .toJSON()

    expect(tree).toMatchSnapshot()
  })
})
