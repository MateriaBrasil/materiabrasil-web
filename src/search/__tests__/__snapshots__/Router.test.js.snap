// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`with /search route renders correctly 1`] = `
<div>
  renderRoot
</div>
`;

exports[`with /search/:term route renders correctly 1`] = `
<div>
  renderList
</div>
`;
